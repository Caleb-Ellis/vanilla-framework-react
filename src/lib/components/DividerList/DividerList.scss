@import 'vanilla-framework/scss/base';
@import 'vanilla-framework/scss/patterns_divider';
@import 'vanilla-framework/scss/patterns_grid';
@import 'vanilla-framework/scss/grid/shelves-grid';
@include vf-b-typography;
@include vf-p-divider;
@include vf-p-grid;
@include vf-p-grid-modifications;