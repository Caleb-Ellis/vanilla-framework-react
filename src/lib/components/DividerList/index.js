export { default as DividerList } from './DividerList';
export { default as DividerListItem } from './DividerListItem';
