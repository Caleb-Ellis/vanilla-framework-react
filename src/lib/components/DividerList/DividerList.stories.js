import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import DividerList from './DividerList';
import DividerListItem from './DividerListItem';

storiesOf('Divider List', module)
  .add('Two Children',
    withInfo('A DividerList component is used to separate related content items. The size of each divider is determined by the equation 12 / # of DividerListItems (12 / 2 = col-6).')(() => (
      <DividerList>
        <DividerListItem>
          <h2>Lorem ipsum</h2>
          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quasi, labore at suscipit
          necessitatibus cumque commodi velit.
          </p>
        </DividerListItem>
        <DividerListItem>
          <h2>Cumque commodi</h2>
          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quasi, labore at suscipit
            necessitatibus cumque commodi velit.
          </p>
        </DividerListItem>
      </DividerList>),
    ),
  )

  .add('Three Children',
    withInfo('A DividerList component is used to separate related content items. The size of each divider is determined by the equation 12 / # of DividerListItems (12 / 3 = col-4).')(() => (
      <DividerList>
        <DividerListItem>
          <h2>Lorem ipsum</h2>
          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quasi, labore at suscipit
          necessitatibus cumque commodi velit.
          </p>
        </DividerListItem>
        <DividerListItem>
          <h2>Dolor sit</h2>
          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quasi, labore at suscipit
            necessitatibus cumque commodi velit. Lorem ipsum dolor sit amet, consectetur adipisicing
            elit. Quasi, labore at suscipit necessitatibus cumque commodi velit. Lorem ipsum dolor
            sit amet, consectetur adipisicing elit. Quasi, labore at suscipit necessitatibus cumque
            commodi velit.
          </p>
        </DividerListItem>
        <DividerListItem>
          <h2>Cumque commodi</h2>
          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quasi, labore at suscipit
            necessitatibus cumque commodi velit.
          </p>
        </DividerListItem>
      </DividerList>),
    ),
  )

  .add('Four Children',
    withInfo('A DividerList component is used to separate related content items. The size of each divider is determined by the equation 12 / # of DividerListItems (12 / 4 = col-3).')(() => (
      <DividerList>
        <DividerListItem>
          <h2>Lorem ipsum</h2>
          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quasi, labore at suscipit
            necessitatibus cumque commodi velit.
          </p>
        </DividerListItem>
        <DividerListItem>
          <h2>Dolor sit</h2>
          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quasi, labore at suscipit
            necessitatibus cumque commodi velit. Lorem ipsum dolor sit amet, consectetur adipisicing
            elit. Quasi, labore at suscipit necessitatibus cumque commodi velit. Lorem ipsum dolor
            sit amet, consectetur adipisicing elit. Quasi, labore at suscipit necessitatibus cumque
            commodi velit.
          </p>
        </DividerListItem>
        <DividerListItem>
          <h2>Cumque commodi</h2>
          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quasi, labore at suscipit
            necessitatibus cumque commodi velit.
          </p>
        </DividerListItem>
        <DividerListItem>
          <h2>Sit amet</h2>
          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor
            incididunt ut labore et dolore magna aliqua.
          </p>
        </DividerListItem>
      </DividerList>),
    ),
  );
