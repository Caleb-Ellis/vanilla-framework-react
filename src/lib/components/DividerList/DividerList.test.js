import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import DividerList from './DividerList';
import DividerListItem from './DividerListItem';

describe('DividerList component', () => {
  it('should render two children correctly', () => {
    const dividerList = ReactTestRenderer.create(
      <DividerList>
        <DividerListItem>
          One
        </DividerListItem>
        <DividerListItem>
          Two
        </DividerListItem>
      </DividerList>,
    );
    const json = dividerList.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render three children correctly', () => {
    const dividerList = ReactTestRenderer.create(
      <DividerList>
        <DividerListItem>
          One
        </DividerListItem>
        <DividerListItem>
          Two
        </DividerListItem>
        <DividerListItem>
          Three
        </DividerListItem>
      </DividerList>,
    );
    const json = dividerList.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render four children correctly', () => {
    const dividerList = ReactTestRenderer.create(
      <DividerList>
        <DividerListItem>
          One
        </DividerListItem>
        <DividerListItem>
          Two
        </DividerListItem>
        <DividerListItem>
          Three
        </DividerListItem>
        <DividerListItem>
          Four
        </DividerListItem>
      </DividerList>,
    );
    const json = dividerList.toJSON();
    expect(json).toMatchSnapshot();
  });
});
