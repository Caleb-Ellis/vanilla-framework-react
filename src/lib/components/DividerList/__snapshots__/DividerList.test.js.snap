// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DividerList component should render four children correctly 1`] = `
<div
  className="p-divider"
>
  <div
    className="col-3 p-divider__block"
  >
    One
  </div>
  <div
    className="col-3 p-divider__block"
  >
    Two
  </div>
  <div
    className="col-3 p-divider__block"
  >
    Three
  </div>
  <div
    className="col-3 p-divider__block"
  >
    Four
  </div>
</div>
`;

exports[`DividerList component should render three children correctly 1`] = `
<div
  className="p-divider"
>
  <div
    className="col-4 p-divider__block"
  >
    One
  </div>
  <div
    className="col-4 p-divider__block"
  >
    Two
  </div>
  <div
    className="col-4 p-divider__block"
  >
    Three
  </div>
</div>
`;

exports[`DividerList component should render two children correctly 1`] = `
<div
  className="p-divider"
>
  <div
    className="col-6 p-divider__block"
  >
    One
  </div>
  <div
    className="col-6 p-divider__block"
  >
    Two
  </div>
</div>
`;
